### PR TITLE
Version initial of vmauth chart

### PR DESCRIPTION
**Pull request by aeciopires**
**What was done?**

* The initial version of the helm chart of the vmauth component was created.
* The repository's primary README.md documentation has been improved.
* Documentation has been created on how others can contribute to the code.
* A page was created with the necessary tools to install / develop the helm charts.
* Documentation on how to install the victoria-metrics-cluster and victoria-metrics-auth helm chart has been improved.

**Why was it done?**

To contribute to the development of the project and to thank for the dedication in the development of the software and to make it open source.

PS: I recommend documenting the values and other information in ``README.md`` using the ``helm-docs`` command and the ``README.md.gotmpl`` template file. It is very practical. See **victoria-metrics-auth** chart.

Ref: https://github.com/norwoodj/helm-docs

### CODE REVIEW

File: charts/victoria-metrics-auth/values.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/values.yaml:45]: vmauth doesnt support remoteWriteUrl

File: charts/victoria-metrics-auth/values.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/values.yaml:152]: i guess, pvc not needed for vmauth

File: charts/victoria-metrics-auth/templates/pvc.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/pvc.yaml:1]: maybe its not needed?

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/deployment.yaml:42]: remoteWrite is not supported.

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/deployment.yaml:68]: is it needed?

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/deployment.yaml:96]: is it needed?

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/deployment.yaml:104]: probably, its better to use secret instead of configmap, WDYT?

File: charts/victoria-metrics-auth/templates/clusterrolebinding.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/clusterrolebinding.yaml:1]: vmauth do not needed this role.

File: charts/victoria-metrics-auth/templates/clusterrole.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/clusterrole.yaml:2]: this permissions is not needed for vmauth.

File: charts/victoria-metrics-auth/templates/configmap.yaml
Comment (f41gh7) [inline charts/victoria-metrics-auth/templates/configmap.yaml:3]: replace with secret ? helm has b64enc func, maybe it will be useful.

**Review by f41gh7 — CHANGES_REQUESTED** [review]
Thanks for contribution, added questions.

File: charts/victoria-metrics-auth/values.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/values.yaml:45]: Thanks for the info @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/values.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/values.yaml:152]: Thanks @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/templates/pvc.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/pvc.yaml:1]: Thanks @f41gh7, I will remove this file in the next commit.

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/deployment.yaml:42]: Thanks @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/deployment.yaml:68]: Thanks @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/deployment.yaml:96]: Thanks @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/templates/clusterrolebinding.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/clusterrolebinding.yaml:1]: Thanks @f41gh7, I will remove this file in the next commit.

File: charts/victoria-metrics-auth/templates/clusterrole.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/clusterrole.yaml:2]: Thanks @f41gh7, I will remove this in the next commit.

File: charts/victoria-metrics-auth/templates/deployment.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/deployment.yaml:104]: Thanks @f41gh7. I changed for secret in next commit.

File: charts/victoria-metrics-auth/templates/configmap.yaml
Comment (aeciopires) [reply charts/victoria-metrics-auth/templates/configmap.yaml:3]: Thanks @f41gh7. I changed for secret in next commit.

**Comment by aeciopires** [comment]
@f41gh7 do you can review again? 
I made the changes and tested it in my production environment and it worked well. But a second review is good before merging

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by tenmozes** [comment]
@aeciopires this is amazing, thank you so much

**Pull request merged**
